sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/components/EmailGenerator.groovy

```diff
@@ -1,48 +1,65 @@
 package life.qbic.samplenotificator.components
 
-import life.qbic.business.subscription.Subscriber
+import life.qbic.business.notification.create.NotificationContent
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
 
 import java.util.concurrent.TimeUnit
 
+
 /**
  * Class responsible for generating an sending an email from the provided notification and subscriber information
  *
  * EmailGenerator converts the notification created in the CreateNotification Use case into an emailBody
  * The contactInformation such as e.g. the emailAddress of the recipient is retrieved from the associated provided Subscriber
  * The email is then send via a commandline call to the mailutils tool(@link <a href=https://mailutils.org/>mailutils</a> via a ProcessBuilder
  *
- * @since: 1.0.0
- *
+ * @since: 1.0.0*
  */
 class EmailGenerator {
 
     private String subject = "Project Update"
-    Map<Subscriber, String> notificationPerSubscriber
+    List<NotificationContent> notificationContentList
+    private InputStream EMAIL_HTML_TEMPLATE_STREAM
+    private EmailHTMLTemplate emailHTMLTemplate
+    private List<Document> filledTemplates = []
 
-    EmailGenerator(Map<Subscriber, String> notificationPerSubscriber) {
-        this.notificationPerSubscriber = notificationPerSubscriber
+    EmailGenerator(List<NotificationContent> notificationContentList) {
+        this.notificationContentList = notificationContentList
     }
 
     /**
      * Triggers the creation and submission of each email
      * containing the provided notifications to the provided subscribers
      */
-    void initializeEmailSubmission(){
-        notificationPerSubscriber.each {
-            //ToDo Replace this with subscriber mail
-            sendEmail("Steffen.greiner@uni-tuebingen.de", it.value)
+    void initializeEmailSubmission() {
+        notificationContentList.each { NotificationContent notificationContent ->
+            accessEmailTemplate()
+            prepareHTMLEmail(notificationContent)
+            //ToDo sendEmail will be handled by a different PR
+            //sendEmail("Steffen.greiner@uni-tuebingen.de", filledTemplate.html())
```

Comment:
ok, keep in mind this should be the email that is provided and not your hardcoded mail. but I understand you did it for testing reasons :)

Reply:
I'll adapt this before merging thanks for reminding me 👍